text/0419-cdk-launchpads.md

```diff
@@ -194,65 +194,88 @@ Those compliance rules are then deployed as
 [CloudFormation Hooks](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks.html),
 which means it's impossible to deploy a CDK application into a given environment that violates any compliance rule defined for it.
 
+When importing the Launchpad into your CDK app (see below),
+the Compliance rules get imported as well,
+meaning they will be evaluated at `synth` time in addition to at deploy time,
+to provide developers the quickest possible feedback loop on whether their apps are compliant or not.
+
 Compliance rules can be defined in JSON/YAML directly:
 
 ```json
 {
   "launchpad": {
     "compliance": {
-      "cfn-guard": {
-        "common": [
-          "let s3_buckets = Resources.[ Type == /S3::Bucket/ ]"
-        ],
-         "rules": {
-           "s3_bucket_name_encryption_check": {
-             "$when": "%s3_buckets !empty",
-           "%s3_buckets": {
-            "Properties": {
-              "BucketName": "== /^MyCompanyPrefix/",
-                "BucketEncryption.ServerSideEncryptionConfiguration[]": {
-                  "ServerSideEncryptionByDefault.SSEAlgorithm": "IN ['aws:KMS']"
+      "cfn-guard": [
+        {
+          "common": [
+            "let s3_buckets = Resources.[ Type == /S3::Bucket/ ]"
+          ],
+          "rules": {
+            "s3_bucket_name_encryption_check": {
+              "$when": "%s3_buckets !empty",
+              "%s3_buckets": {
+                "Properties": {
+                  "BucketName": "== /^MyCompanyPrefix/",
+                  "BucketEncryption.ServerSideEncryptionConfiguration[]": {
+                    "ServerSideEncryptionByDefault.SSEAlgorithm": "IN ['aws:KMS']"
+                  }
                 }
               }
             }
           }
         }
-      }
+      ]
+    }
+  }
+}
+```
+
+Or you can reference a file on disk that contains the Guard rules:
+
+```json
+{
+  "launchpad": {
+    "compliance": {
+      "cfn-guard": ["file:my-guard-rules.guard"]
```

Comment:
Perhaps worth mentioning or adding an example of validating input data against multiple data files? CDK creates cloud assembly with potentially multiple templates. Perhaps the default behavior should use cloud assembly directory for CFN Guard's `--data`. `--rules` can be a file or a directory.

Reply:
I feel like that's a little obvious. Yes, the Guard rules will be applied automatically to each template CDK generates 🙂.

Reply:
Even though it makes a total sense, it wasn't obvious to me from the name of the configuration key :). Perhaps rename it to `cfn-guard-rules`, and give an example for accepting either a file or a directory, like CFN Guard does?

```suggestion
      "cfn-guard-rules": ["file:my-guard-rules.guard"] // or "dir:my-guard-rules"
```

Reply:
That's fair! Changed the example (I've kept the `"cfn-guard"` name of the key though - I feel like the "rules" part belongs more on the actual files than on this key, let me know if you strongly disagree 🙂).

Reply:
Thanks! I do feel quite strongly about the key name, because the file or directory names are provided by the user, and can't be enforced :). Neither tye extension is mandatory from what I remember in my last use of CFN Guard.

Reply:
I feel like that's fine - then name of the file should already include the fact that it's a Guard rules file.

Reply:
The "should" part is not enforceable by any means, but I will stop nudging you on this :)

Reply:
Sounds like a plan! 😄

Reply:
Having these files in s3 or codecommit would be nice for when this is centrally managed w/ control tower (hopefully).  codecommit://my-central-repo/permission-name/ could be a directory for example, obviously roles for access complicate this but central storage pointing to the base set of OU wide rules would be nice.  Perhaps how I'm thinking quickly evolves into a custom rule in config that prevents CF updates that don't meet the cfn guard rules? As I'm typing I think this is going beyond the scope of your goal (?).